packages/pigeon/lib/dart_generator.dart

```diff
@@ -8,13 +8,45 @@ import 'generator_tools.dart';
 /// Options that control how Dart code will be generated.
 class DartOptions {
   /// Constructor for DartOptions.
-  DartOptions({this.isNullSafe = true, this.copyrightHeader});
+  const DartOptions({this.isNullSafe = true, this.copyrightHeader});
 
   /// Determines if the generated code has null safety annotations (Dart >=2.12 required).
-  bool isNullSafe;
+  final bool isNullSafe;
 
   /// A copyright header that will get prepended to generated code.
-  Iterable<String>? copyrightHeader;
+  final Iterable<String>? copyrightHeader;
+
+  /// Creates a [DartOptions] from a Map representation where:
+  /// `x = DartOptions.fromMap(x.toMap())`.
+  static DartOptions fromMap(Map<String, Object> map) {
+    return DartOptions(
+      isNullSafe:
+          // ignore: avoid_bool_literals_in_conditional_expressions
+          map.containsKey('isNullSafe') ? (map['isNullSafe'] as bool?)! : true,
+      copyrightHeader: map.containsKey('copyrightHeader')
+          ? map['copyrightHeader'] as Iterable<String>?
+          : null,
```

Comment:
Isn't this whole expression just `map['copyrightHeader'] as Iterable<String>?`?

Reply:
Yep, sometimes implementations of associative collections throw exceptions when accessing fields that haven't been set, looks like that's not the case for Dart, removed.